src/utils/validation-utils.ts

```diff
@@ -1,8 +1,103 @@
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
 import { Base64 } from 'js-base64';
 import { attempt, isError } from 'lodash-es';
 import * as yup from 'yup';
 import { ImagePullSecretType, SecretTypeDropdownLabel, SourceSecretType } from '../types';
 
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export const hhmmRegex = /^([01]\d|2[0-3]):([0-5]\d)$/;
+
+export const bannerConfigYupSchema = yup.object({
+  summary: yup
+    .string()
+    .required()
+    .min(5, 'Must be at least 5 characters')
+    .max(500, 'Must be at most 500 characters'),
+
+  type: yup.mixed<'info' | 'warning' | 'danger'>().oneOf(['info', 'warning', 'danger']).required(),
+
+  repeatType: yup
```

Comment:
Do we need this repeatType when we know that if dayOfWeek is provided it is weekly and if dayOfMonth is provided without month and year it is monthly.

Reply:
updated.